dll/win32/shell32/shlexec.cpp

```diff
@@ -2332,26 +2317,19 @@ static BOOL SHELL_execute(LPSHELLEXECUTEINFOW sei, SHELL_ExecuteW32 execfunc)
         /* if so, prefix lpFile with http:// and call ShellExecute */
         WCHAR lpstrTmpFile[256];
         strcpyW(lpstrTmpFile, L"http://");
-        strcatW(lpstrTmpFile, lpFile);
+        strcatW(lpstrTmpFile, lpFile); // FIXME: Possible buffer overflow
```

Comment:
Lol, calling "lpstr" a fixed-sized string buffer...

Two suggestions:
1. Either:
```suggestion
        WCHAR lpstrTmpFile[256]; // TODO: Rename to wszTmpFile
        HRESULT hr = StringCchCopyW(lpstrTmpFile, _countof(lpstrTmpFile), L"http://");
        if (SUCCEEDED(hr))
            hr = StringCchCatW(lpstrTmpFile, _countof(lpstrTmpFile), lpFile);
        if (!SUCCEEDED(hr))
            return (sei->hInstApp = SE_ERR_OOM); // FIXME: Should we set hInstApp in case of failure?
```
2. Or:
```suggestion
        CHeapPtr<WCHAR, CLocalAllocator> lpstrTmpFile; // TODO: Rename to wszTmpFile
        if (!lpstrTmpFile.Allocate(wcslen(L"http://") + wcslen(lpFile) + 1);
            return (sei->hInstApp = SE_ERR_OOM); // FIXME: Should we set hInstApp in case of failure?
        HRESULT hr = StringCchCopyW(lpstrTmpFile, _countof(lpstrTmpFile), L"http://");
        assert(SUCCEEDED(hr));
        hr = StringCchCatW(lpstrTmpFile, _countof(lpstrTmpFile), lpFile);
        assert(SUCCEEDED(hr));
```

Reply:
This code has nothing to do with this PR in the first place, I just spotted it and I feel it at least needs a comment.

Reply:
ok so let's not touch it there.